### ISSUE TITLE
Keepassx Not Responding with 1.8.0

### ISSUE THREAD

**Issue by ghost** [QupZilla/qupzilla#1484]
Hi,

I'm using Slack 14.1 x86_64
QT5 5.2.1
WebKit version 538.1
keepassx-0.4.3

When using Keepassx and pressing my key shortcut I assigned to it, nothing happens, my username and password will not get filled in and auto completed on any site I've tried.

When I compiled QupZilla against QT4 which gave me  534.34 it was working.

**Comment by moulei** [comment]
I was told on this report below that the shortcut key issues is on non-latin keyboard layouts and that is has nothing to do with US keyboard layouts;

https://bugreports.qt-project.org/browse/QTBUG-32274#comment-268986

I'm using a US keyboard layout...

**Comment by JHooverman** [comment]
@moulei: I did the following to reproduce your issue (please let me know, when I am on the wrong way) with keepassx-2.0 alpha 6:

* Set Global Auto-Type Shortcut to CRTL+z via `Extras -> Settings -> General -> Global Auto-Type Shortcut`
* Check `Enable Auto-Type for this entry` in a test account, here tumblr
* Verify, that QupZilla is up and running
* Press `+` to add a window title for applying CRTL (should be QupZilla), see image

![kp](http://i.imgur.com/Evxy4u2.png)

Now, keepassx does not offer an entry for QupZilla to be selected. Qt does currently not set the [WM_NAME property](https://bugreports.qt-project.org/browse/QTBUG-42209). This might be the reason, that auto-completion will not work.

**Comment by moulei** [comment]
I'm using 0.4.3 

The first screenshot is like 0.4.3 

The second seems to be like the ```Tools``` entry in 0.4.3 where you can pick either of these settings;
```
Auto-Type: Customize Sequence
Auto-Type: Select Target Window
```
So if the second screenshot is for the ```Select Target Window``` I rarely have to use this in Firefox with KeePassX, as example, maybe out of 50 sites, I only use it on 1-2...

**Comment by JHooverman** [comment]
In the meantime I did some tests and I found out, that the KeePassX global auto-type only works as you already stated with QupZilla compiled against qt4. Furthermore, this seems to work only with KeePassX-0.4.3 and **not** with version 2.0 alpha6 (tested with openSUSE Tumbleweed only). 

The global auto-type option works as follows: If you enter in the browser window the combination of the global hot keys from the keyboard, KeePassX searches the entire password database for a matching entry. KeePassX takes the **title of the currently opened window (eg Firefox)** and then searches all records in the database for a matching title line entry. 

This is exactly the reason why this does not work in QupZilla compiled against Qt5: Qt5 does currently not set the [WM_NAME property](https://bugreports.qt.io/browse/QTBUG-42209). The global auto-type option supposes to work, once this Qt5 bug is resolved.

**Comment by moulei** [comment]
I've waited a year to use QupZilla, oh well ANOTHER FLIPPIN QT BUG! :(

Errrr

THANKS

**Comment by The-Compiler** [comment]
Keep in mind this also was fixed in KeePassX, so you can as well update that (or use a git version): https://www.keepassx.org/dev/issues/236

**Comment by JHooverman** [comment]
@The-Compiler: Do I understand correctly, that using a current keepassx version from git will make the global auto-type option work with Qt5?

**Comment by The-Compiler** [comment]
I believe so, yes. Though I haven't tested it.

Basically there a `WM_NAME` atom (window title in an essentially unknown encoding), and `_NET_WM_NAME` (window title in UTF-8). The older standard ([ICCM](http://tronche.com/gui/x/icccm/sec-4.html#s-4.1.2.1)) does not explicitely require `WM_NAME`, and the newer standard ([EWMH](http://standards.freedesktop.org/wm-spec/wm-spec-1.3.html#idm140130317621408)) doesn't say anything about `WM_NAME` when `_NET_WM_NAME` is set. Due to a [wrong comparison](https://www.keepassx.org/dev/projects/keepassx/repository/revisions/870d7355cabdd4ad4445bf4655965893db826f81/diff/src/autotype/x11/AutoTypeX11.cpp) this case was handled incorrectly in KeePassX.

So basically this is *not* a Qt bug, but a KeePassX bug - arguably Qt should set `WM_NAME` to improve interoperability, but it's not required to do that.

**Comment by JHooverman** [comment]
Thank you for your explanation. I will compile a keepassx git-revision and report back back the result.

**Comment by JHooverman** [comment]
@moulei: It works!

Tested with:
Application version: 1.9.0 (c1901cd81d)
Qt version: 5.4.1
WebKit version: 538.1
Build time: Dec 28 2014 15:49:07
Platform: openSUSE Tumbleweed (Linux)

KeePassX: Latest revision from git (2.0 alpha 6 git revision 835c5411)

@The-Compiler: Thanks again!

**Comment by moulei** [comment]
@The-Compiler: Thanks

I am a geek but even I stay away from Alpha in Linux, beta apps are as far as I go...

But yes as they keep telling us Geeks, Alpha/Beta isnt the same in Windows as Unix... :)

@JHooverman did you also test this with QupZilla 1.8.5?

KeePassX 2 hasn't had an update in 9 months to either another Alpha or switching to Beta version...

Ahh roadmap shows 88% of the way to the first beta;

https://www.keepassx.org/dev/projects/keepassx/roadmap

**Comment by The-Compiler** [comment]
@moulei Another possibility would probably be to build whatever version you're running currently yourself, and apply [the patch](https://www.keepassx.org/dev/projects/keepassx/repository/revisions/870d7355cabdd4ad4445bf4655965893db826f81/diff/src/autotype/x11/AutoTypeX11.cpp?format=diff) on top of that.

**Comment by moulei** [comment]
Ok thanks

**Comment by moulei** [comment]
Well I was hoping 1.8.6 might work with KeePassX 0.4.3, but no such luck...

Is it possible to get it working with 0.4.3? I don't want to use a Alpha version of KeePassX and it would be great to get this working with the Stable version, until 2 is released final...

**Comment by The-Compiler** [comment]
I don't think there's anything QupZilla can do about this - you will have to wait for a Qt fix, or update KeePassX, or apply that patch on your stable KeePassX.

**Comment by moulei** [comment]
What that .diff patch works for 0.4.3?

**Comment by The-Compiler** [comment]
I don't know, but it's worth a try ;)

**Comment by moulei** [comment]
Ahh I looked at the src and ```/src/autotype/x11/AutoTypeX11.cpp``` doesn't exist in 0.4.3, so that takes care of that...

**Comment by JHooverman** [comment]
Solved in Qt-5.5.0 beta
